src/components/NacelleReferences/NacelleReferences.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { useEffect, useState } from 'react'
+import { BaseExtensionSDK } from 'contentful-ui-extensions-sdk'
+import Loading from './Loading'
+import ReferenceTable, { ReferenceTableRow } from './ReferenceTable'
+import getNacelleFields from '../../utils/getNacelleFields'
+
+interface NacelleReferencesProps {
+  sdk: BaseExtensionSDK
+}
+
+const NacelleReferences = (props: NacelleReferencesProps) => {
+  const [data, setData] = useState<ReferenceTableRow[]>([])
+  const [loading, setLoading] = useState(true)
+
+  useEffect(() => {
+    const cmaAdapter = props.sdk.cmaAdapter
+    const appId = props.sdk.ids.app
+    const spaceId = props.sdk.ids.space
+    const envId = props.sdk.ids.environment
+
+    getNacelleFields(cmaAdapter, appId, spaceId, envId)
+      .then((res) => {
+        setData(res)
+        setLoading(false)
+      })
+      .catch((err) => {
+        throw new Error(err)
+      })
+  }, [props.sdk])
```

Comment:
Hey @NWRichmond I think this suggestion is what is causing the infinite loop in the app page for you: 
referencing `loading` inside `useEffects` triggers `useEffects` to reference `loading` and so on

I think having the page loading as a default isn't an issue because we aren't just loading the data fetching but everything that happens in `useEffects` and since `useEffects` will be called very early in the lifecycles it should always resolve `loading` if we add that `finally` piece

Reply:
Working as expected now ✅

![Contentful app references collectionGrid](https://github.com/getnacelle/contentful-app-nacelle/assets/5732000/b8da5139-e121-4991-8381-6efc6441ea60)